pytorch_lightning/core/optimizer.py

```diff
@@ -207,6 +207,7 @@ def closure_dis():
             profiler_name = f"optimizer_step_and_closure_{self._optimizer_idx}"
 
         self.__optimizer_step(*args, closure=closure, profiler_name=profiler_name, **kwargs)
+        self._trainer.fit_loop.epoch_loop.batch_loop.optim_progress.optimizer.step.increment_processed()
```

Comment:
This assumes a rigid loop structure. Is there maybe a way to do this more dynamically? Ideally the trainer (and optimizer, lightningmodule etc.) only know about the entry point to the top-most loop and everything else is handled by the loops.

thoughts on how to handle this @awaelchli @ananthsub @carmocca  ?

Reply:
Moved to batch_loop.